core/src/main/java/org/elasticsearch/index/translog/BufferingTranslogWriter.java

```diff
@@ -102,20 +114,28 @@ public boolean syncNeeded() {
     }
 
     @Override
-    public void sync() throws IOException {
-        if (!syncNeeded()) {
-            return;
-        }
-        synchronized (this) {
+    public synchronized void sync() throws IOException {
+        if (syncNeeded()) {
+            ensureOpen(); // this call gives a better exception that the incRef if we are closed by a tragic event
             channelReference.incRef();
             try {
+                final long offsetToSync;
+                final int opsCounter;
                 try (ReleasableLock lock = writeLock.acquire()) {
                     flush();
-                    lastSyncedOffset = totalOffset;
+                    offsetToSync = totalOffset;
+                    opsCounter = operationCounter;
                 }
                 // we can do this outside of the write lock but we have to protect from
                 // concurrent syncs
-                checkpoint(lastSyncedOffset, operationCounter, channelReference);
+                try {
+                    ensureOpen(); // just for kicks - the checkpoint happens or not either way
+                    checkpoint(offsetToSync, opsCounter, channelReference);
+                } catch (IOException ex) {
```

Comment:
Hmm should we catch `Throwable` instead of `IOException` like we do in the other tragic cases...?  But I guess then we'd need to exclude `AlreadyClosedException` (that's not tragic)?